<!DOCTYPE html>
<html>

<head><meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<meta http-equiv="Accept-CH" content="DPR, Viewport-Width, Width">
<link rel="icon" href=/dna.png type="image/gif">


<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
<link rel="preload"
      as="style"
      href="https://fonts.googleapis.com/css2?family=Alata&family=Lora:ital,wght@0,400;0,500;0,600;0,700;1,400;1,500;1,600;1,700&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap"
>
<link rel="stylesheet"
      href="https://fonts.googleapis.com/css2?family=Alata&family=Lora:ital,wght@0,400;0,500;0,600;0,700;1,400;1,500;1,600;1,700&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap"
      media="print" onload="this.media='all'" />
<noscript>
  <link
          href="https://fonts.googleapis.com/css2?family=Alata&family=Lora:ital,wght@0,400;0,500;0,600;0,700;1,400;1,500;1,600;1,700&family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap"
          rel="stylesheet">
</noscript>


<link rel="stylesheet" href="/css/font.css" media="all">



<meta property="og:title" content="My DESeq2 Workflow" />
<meta property="og:description" content="Differential Gene Expression I use this code to run DESeq2 for evey differential gene expression analysis I have:
library(DESeq2) library(biomaRt) library(DEGreport) library(ggplot2) library(dplyr) # Load data setwd(&#39;/home/&lt;directory&gt;&#39;) cts &lt;- read.delim(&#39;counts_data.csv&#39;, sep=&#39;,&#39;, header = TRUE, row.names = 1) condition &lt;- factor(c(&#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;)) coldata &lt;- data.frame(row.names = colnames(cts), condition) Groups &lt;- make.names(c(&#34;Control&#34;, &#34;Disease&#34;)) keep &lt;- rowSums(cts &gt;= 10) &gt;= min(table(coldata)) cts &lt;- cts[keep, ] dds &lt;- DESeqDataSetFromMatrix(countData = round(cts), colData = coldata, design = ~condition) dds &lt;- DESeq(dds) # Save all the results res &lt;- results(dds, contrast = c(&#34;condition&#34;, Groups[2], Groups[1]), alpha = 0." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://rahriver.github.io/posts/deseq2/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2023-08-27T19:07:04+03:30" />
<meta property="article:modified_time" content="2023-08-27T19:07:04+03:30" /><meta property="og:site_name" content="Portfolio" />

<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="My DESeq2 Workflow"/>
<meta name="twitter:description" content="Differential Gene Expression I use this code to run DESeq2 for evey differential gene expression analysis I have:
library(DESeq2) library(biomaRt) library(DEGreport) library(ggplot2) library(dplyr) # Load data setwd(&#39;/home/&lt;directory&gt;&#39;) cts &lt;- read.delim(&#39;counts_data.csv&#39;, sep=&#39;,&#39;, header = TRUE, row.names = 1) condition &lt;- factor(c(&#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Control&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;, &#34;Disease&#34;)) coldata &lt;- data.frame(row.names = colnames(cts), condition) Groups &lt;- make.names(c(&#34;Control&#34;, &#34;Disease&#34;)) keep &lt;- rowSums(cts &gt;= 10) &gt;= min(table(coldata)) cts &lt;- cts[keep, ] dds &lt;- DESeqDataSetFromMatrix(countData = round(cts), colData = coldata, design = ~condition) dds &lt;- DESeq(dds) # Save all the results res &lt;- results(dds, contrast = c(&#34;condition&#34;, Groups[2], Groups[1]), alpha = 0."/>


<link rel="stylesheet" href="/bootstrap-5/css/bootstrap.min.css" media="all"><link rel="stylesheet" href="/css/header.css" media="all">
<link rel="stylesheet" href="/css/footer.css" media="all">


<link rel="stylesheet" href="/css/theme.css" media="all">




<style>
    :root {
        --text-color: #343a40;
        --text-secondary-color: #6c757d;
        --background-color: #eaedf0;
        --secondary-background-color: #64ffda1a;
        --primary-color: #007bff;
        --secondary-color: #f8f9fa;

         
        --text-color-dark: #e4e6eb;
        --text-secondary-color-dark: #b0b3b8;
        --background-color-dark: #18191a;
        --secondary-background-color-dark: #212529;
        --primary-color-dark: #ffffff;
        --secondary-color-dark: #212529;
    }
    body {
        font-size: 1rem;
        font-weight: 400;
        line-height: 1.5;
        text-align: left;
    }

    html {
        background-color: var(--background-color) !important;
    }

    body::-webkit-scrollbar {
        width: .5em;
        height: .5em;
        background-color: var(--background-color);
    }
    
    ::-webkit-scrollbar-track {
        box-shadow: inset 0 0 6px var(--background-color);
        border-radius: 1rem;
    }
    
    ::-webkit-scrollbar-thumb {
        border-radius: 1rem;
        background-color: var(--secondary-color);
        outline: 1px solid var(--background-color);
    }

    #search-content::-webkit-scrollbar {
        width: .5em;
        height: .1em;
        background-color: var(--background-color);
    }
</style>

<meta name="description" content="">
<link rel="stylesheet" href="/css/single.css">


<script defer src="/fontawesome-5/all-5.15.4.js"></script>

  <title>
My DESeq2 Workflow | rahriver

  </title>
</head>

<body class="light">
  
  
<script>
    let localStorageValue = localStorage.getItem("pref-theme");
    let mediaQuery = window.matchMedia('(prefers-color-scheme: dark)').matches;

    switch (localStorageValue) {
        case "dark":
            document.body.classList.add('dark');
            break;
        case "light":
            document.body.classList.remove('dark');
            break;
        default:
            if (mediaQuery) {
                document.body.classList.add('dark');
            }
            break;
    }
</script>



<header>
    <nav class="pt-3 navbar navbar-expand-lg animate">
        <div class="container-fluid mx-xs-2 mx-sm-5 mx-md-5 mx-lg-5">
            
            <a class="navbar-brand primary-font text-wrap" href="/">
                
                <img src="/dna.png" width="30" height="30"
                    class="d-inline-block align-top">
                rahriver
                
            </a>

            

            
            <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarContent"
                aria-controls="navbarContent" aria-expanded="false" aria-label="Toggle navigation">
                <svg aria-hidden="true" height="24" viewBox="0 0 16 16" version="1.1" width="24" data-view-component="true">
                    <path fill-rule="evenodd" d="M1 2.75A.75.75 0 011.75 2h12.5a.75.75 0 110 1.5H1.75A.75.75 0 011 2.75zm0 5A.75.75 0 011.75 7h12.5a.75.75 0 110 1.5H1.75A.75.75 0 011 7.75zM1.75 12a.75.75 0 100 1.5h12.5a.75.75 0 100-1.5H1.75z"></path>
                </svg>
            </button>

            
            <div class="collapse navbar-collapse text-wrap primary-font" id="navbarContent">
                <ul class="navbar-nav ms-auto text-center">
                    

                    
                    <li class="nav-item navbar-text">
                        <a class="nav-link" href="/#about" aria-label="about">
                            &lt;/&gt; About Me
                        </a>
                    </li>
                    

                    
                    <li class="nav-item navbar-text">
                        <a class="nav-link" href="/#experience"
                            aria-label="experience">
                            üíº Work Experiences
                        </a>
                    </li>
                    

                    
                    <li class="nav-item navbar-text">
                        <a class="nav-link" href="/#education"
                            aria-label="education">
                            üéì Education
                        </a>
                    </li>
                    

                    
                    <li class="nav-item navbar-text">
                        <a class="nav-link" href="/#projects"
                            aria-label="projects">
                            ‚öôÔ∏è Projects
                        </a>
                    </li>
                    

                    
                    <li class="nav-item navbar-text">
                        <a class="nav-link" href="/#achievements"
                            aria-label="achievements">
                            üìÑ Publications
                        </a>
                    </li>
                    

                    

                    
                    
                    
                    
                    <li class="nav-item navbar-text">
                        <a class="nav-link" href="/posts" title="Blog posts">
                            
                            üì∞ Blog
                        </a>
                    </li>
                    
                    

                    
                    <li class="nav-item navbar-text">
                        
                        <div class="text-center">
                            <button id="theme-toggle">
                                <svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                                    <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
                                </svg>
                                <svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                                    <circle cx="12" cy="12" r="5"></circle>
                                    <line x1="12" y1="1" x2="12" y2="3"></line>
                                    <line x1="12" y1="21" x2="12" y2="23"></line>
                                    <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                                    <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                                    <line x1="1" y1="12" x2="3" y2="12"></line>
                                    <line x1="21" y1="12" x2="23" y2="12"></line>
                                    <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                                    <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                                </svg>
                            </button>
                        </div>
                    </li>
                    

                </ul>

            </div>
        </div>
    </nav>
</header>
<div id="content">
<section id="single">
  <div class="container">
    <div class="row justify-content-center">
      <div class="col-sm-12 col-md-12 col-lg-9">
        <div class="pr-lg-4">
          <div class="title mb-5">
            <h1 class="text-center mb-4">My DESeq2 Workflow</h1>
            <div class="text-center">
              Ramin 
              <small>|</small>
              Aug 27, 2023

              
              <span id="readingTime">
                min read
              </span>
              
            </div>
          </div>
          
          <div class="featured-image">
            <img class="img-fluid" src="/images/posts/deseq.jpeg" alt="My DESeq2 Workflow">
          </div>
          
          <article class="page-content  p-2">
          <h1 id="differential-gene-expression">Differential Gene Expression</h1>
<p>I use this code to run DESeq2 for evey differential gene expression analysis I have:</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(DESeq2)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(biomaRt)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(DEGreport)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(ggplot2)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(dplyr)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Load data</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">setwd</span>(<span style="color:#e6db74">&#39;/home/&lt;directory&gt;&#39;</span>)
</span></span><span style="display:flex;"><span>cts <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">read.delim</span>(<span style="color:#e6db74">&#39;counts_data.csv&#39;</span>, sep<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;,&#39;</span>, header <span style="color:#f92672">=</span> <span style="color:#66d9ef">TRUE</span>, row.names <span style="color:#f92672">=</span> <span style="color:#ae81ff">1</span>)
</span></span><span style="display:flex;"><span>condition <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">factor</span>(<span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>))
</span></span><span style="display:flex;"><span>coldata <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">data.frame</span>(row.names <span style="color:#f92672">=</span> <span style="color:#a6e22e">colnames</span>(cts), condition)
</span></span><span style="display:flex;"><span>Groups <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">make.names</span>(<span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;Control&#34;</span>, <span style="color:#e6db74">&#34;Disease&#34;</span>))
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>keep <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">rowSums</span>(cts <span style="color:#f92672">&gt;=</span> <span style="color:#ae81ff">10</span>) <span style="color:#f92672">&gt;=</span> <span style="color:#a6e22e">min</span>(<span style="color:#a6e22e">table</span>(coldata))
</span></span><span style="display:flex;"><span>cts <span style="color:#f92672">&lt;-</span> cts[keep, ]
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>dds <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">DESeqDataSetFromMatrix</span>(countData <span style="color:#f92672">=</span> <span style="color:#a6e22e">round</span>(cts),
</span></span><span style="display:flex;"><span>                              colData <span style="color:#f92672">=</span> coldata,
</span></span><span style="display:flex;"><span>                              design <span style="color:#f92672">=</span> <span style="color:#f92672">~</span>condition)
</span></span><span style="display:flex;"><span>dds <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">DESeq</span>(dds)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Save all the results</span>
</span></span><span style="display:flex;"><span>res <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">results</span>(dds, contrast <span style="color:#f92672">=</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;condition&#34;</span>, Groups[2], Groups[1]),
</span></span><span style="display:flex;"><span>               alpha <span style="color:#f92672">=</span> <span style="color:#ae81ff">0.05</span>,
</span></span><span style="display:flex;"><span>               pAdjustMethod <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;fdr&#34;</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>res <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as.data.frame</span>(res)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.csv</span>(res, <span style="color:#e6db74">&#39;all_results.csv&#39;</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Filtered DEGs</span>
</span></span><span style="display:flex;"><span>res <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">subset</span>(res, padj <span style="color:#f92672">&lt;</span> <span style="color:#ae81ff">0.01</span> <span style="color:#f92672">&amp;</span> <span style="color:#a6e22e">abs</span>(log2FoldChange) <span style="color:#f92672">&gt;</span> <span style="color:#ae81ff">1</span>)
</span></span><span style="display:flex;"><span>res <span style="color:#f92672">&lt;-</span> res<span style="color:#a6e22e">[order</span>(res<span style="color:#f92672">$</span>log2FoldChange), ]
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.csv</span>(res, <span style="color:#e6db74">&#39;deg_results.csv&#39;</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Summary of DEGs</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">summary</span>(res)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Top 10 genes: downregulated and upregulated</span>
</span></span><span style="display:flex;"><span>up <span style="color:#f92672">&lt;-</span> res<span style="color:#a6e22e">[order</span>(res<span style="color:#f92672">$</span>log2FoldChange)[1<span style="color:#f92672">:</span><span style="color:#ae81ff">10</span>], ]
</span></span><span style="display:flex;"><span>down <span style="color:#f92672">&lt;-</span> res<span style="color:#a6e22e">[order</span>(<span style="color:#f92672">-</span>res<span style="color:#f92672">$</span>log2FoldChange)[1<span style="color:#f92672">:</span><span style="color:#ae81ff">10</span>], ]
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.csv</span>(up, <span style="color:#e6db74">&#39;DEG_top10_up.csv&#39;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.csv</span>(down, <span style="color:#e6db74">&#39;DEG_top10_down.csv&#39;</span>)
</span></span></code></pre></div><h2 id="ensembl-ids-to-gene-symbols">ENSEMBL IDs to Gene Symbols</h2>
<p>Sometimes we have ENSEMBL IDs in the original counts matrix, but for plots and further analysis we need the official gene symbols. I use the <code>bioMart</code> package to convert:</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span>gene_id <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">rownames</span>(res)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Choose the database</span>
</span></span><span style="display:flex;"><span>ensemble <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">useEnsembl</span>(biomart<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;genes&#39;</span>)
</span></span><span style="display:flex;"><span>datasets <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">listDatasets</span>(ensemble)
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Select the species: Homo sapiens</span>
</span></span><span style="display:flex;"><span>ensemble <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">useDataset</span>(dataset<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;hsapiens_gene_ensembl&#34;</span>, mart<span style="color:#f92672">=</span>ensemble)
</span></span><span style="display:flex;"><span>attributes <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">listAttributes</span>(ensemble)
</span></span><span style="display:flex;"><span>ensemble <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">useEnsembl</span>(biomart <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;genes&#34;</span>, dataset<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;hsapiens_gene_ensembl&#34;</span>)
</span></span><span style="display:flex;"><span>results <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getBM</span>(mart<span style="color:#f92672">=</span>ensemble, attributes <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;hgnc_symbol&#34;</span>,
</span></span><span style="display:flex;"><span>                 filters <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;ensembl_gene_id&#34;</span>,
</span></span><span style="display:flex;"><span>                 values <span style="color:#f92672">=</span> gene_id)
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Get the corresponding gene symbols</span>
</span></span><span style="display:flex;"><span>gene_data <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getBM</span>(attributes <span style="color:#f92672">=</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;ensembl_gene_id&#34;</span>, <span style="color:#e6db74">&#34;hgnc_symbol&#34;</span>),
</span></span><span style="display:flex;"><span>                   filters <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;ensembl_gene_id&#34;</span>,
</span></span><span style="display:flex;"><span>                   values <span style="color:#f92672">=</span> gene_id,
</span></span><span style="display:flex;"><span>                   mart <span style="color:#f92672">=</span> ensemble)
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Create a dataframe with Ensembl IDs and gene symbols</span>
</span></span><span style="display:flex;"><span>genes <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">data.frame</span>(Ensembl_ID <span style="color:#f92672">=</span> gene_data<span style="color:#f92672">$</span>ensembl_gene_id,
</span></span><span style="display:flex;"><span>                    Gene_Symbol <span style="color:#f92672">=</span> gene_data<span style="color:#f92672">$</span>hgnc_symbol)
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Gene symbols that are not found in the ensembl dataframe</span>
</span></span><span style="display:flex;"><span>not_found <span style="color:#f92672">&lt;-</span> gene_data[gene_data<span style="color:#f92672">$</span>hgnc_symbol <span style="color:#f92672">==</span> <span style="color:#e6db74">&#34;&#34;</span>,]
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">cat</span>(<span style="color:#e6db74">&#34;biomart couldn&#39;t find&#34;</span>, <span style="color:#a6e22e">length</span>(<span style="color:#a6e22e">rownames</span>(not_found)), <span style="color:#e6db74">&#34;of the genes!&#34;</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Remove rows that have not been founded</span>
</span></span><span style="display:flex;"><span>genes <span style="color:#f92672">&lt;-</span> genes[genes<span style="color:#f92672">$</span>Gene_Symbol <span style="color:#f92672">!=</span> <span style="color:#e6db74">&#34;&#34;</span>,]
</span></span><span style="display:flex;"><span>gene_symbols <span style="color:#f92672">&lt;-</span> genes<span style="color:#f92672">$</span>Gene_Symbol
</span></span><span style="display:flex;"><span>gene_ensmbl <span style="color:#f92672">&lt;-</span> genes<span style="color:#f92672">$</span>Ensembl_ID
</span></span><span style="display:flex;"><span>matching_genes <span style="color:#f92672">&lt;-</span> res<span style="color:#a6e22e">[rownames</span>(res) <span style="color:#f92672">%in%</span> gene_ensmbl, ]
</span></span><span style="display:flex;"><span>matching_log2foldchanges <span style="color:#f92672">&lt;-</span> matching_genes<span style="color:#f92672">$</span>log2FoldChange
</span></span><span style="display:flex;"><span>result_df <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">data.frame</span>(GeneSymbol <span style="color:#f92672">=</span> gene_symbols, log2FoldChange <span style="color:#f92672">=</span> matching_log2foldchanges)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.csv</span>(result_df, <span style="color:#e6db74">&#39;genes_log2.csv&#39;</span>, row.names <span style="color:#f92672">=</span> <span style="color:#66d9ef">FALSE</span>)
</span></span></code></pre></div><h2 id="plots">Plots</h2>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-r" data-lang="r"><span style="display:flex;"><span><span style="color:#75715e">#&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;________PLOTS__________&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;#</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Important genes that we want to plot</span>
</span></span><span style="display:flex;"><span>imp_genes <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;CDON&#34;</span>, <span style="color:#e6db74">&#34;OMG&#34;</span>, <span style="color:#e6db74">&#34;TSPAN15&#34;</span>, <span style="color:#e6db74">&#34;CHEK1&#34;</span>, <span style="color:#e6db74">&#34;ROBO2&#34;</span>, <span style="color:#e6db74">&#34;PGA5&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Select the important genes from the dataframe</span>
</span></span><span style="display:flex;"><span>selected <span style="color:#f92672">&lt;-</span> genes <span style="color:#f92672">%&gt;%</span> <span style="color:#a6e22e">filter</span>(Gene_Symbol <span style="color:#f92672">%in%</span> imp_genes)
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Slice the rows (ensembl ids) to give them to the degPlot function so it plots the important genes</span>
</span></span><span style="display:flex;"><span>sel_rows <span style="color:#f92672">&lt;-</span> selected[[1]]
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/table_log2fc_abundance_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">degPlot</span>(dds, genes <span style="color:#f92672">=</span> sel_rows, xs <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;condition&#34;</span>, group <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;condition&#34;</span>, color <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;Accent&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/deg_Plot_Wide.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">degPlotWide</span>(dds, <span style="color:#a6e22e">rownames</span>(dds)[1<span style="color:#f92672">:</span><span style="color:#ae81ff">5</span>], group<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;condition&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Individual Gene Count</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Loop, find expression, plot, save png to folder</span>
</span></span><span style="display:flex;"><span>gene_ids_data <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">read.csv</span>(<span style="color:#e6db74">&#34;gene_ids_plot.csv&#34;</span>)
</span></span><span style="display:flex;"><span>gene_ids <span style="color:#f92672">&lt;-</span> gene_ids_data<span style="color:#f92672">$</span>gene_id
</span></span><span style="display:flex;"><span>gene_symbols <span style="color:#f92672">&lt;-</span> gene_ids_data<span style="color:#f92672">$</span>gene_symbol  <span style="color:#75715e"># Assuming the column name is &#34;gene_symbol&#34;</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">## Loop through gene_ids and create plots</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">for </span>(i in <span style="color:#a6e22e">seq_along</span>(gene_ids)) {
</span></span><span style="display:flex;"><span>        gene_id <span style="color:#f92672">&lt;-</span> gene_ids[i]
</span></span><span style="display:flex;"><span>        gene_symbol <span style="color:#f92672">&lt;-</span> gene_symbols[i]
</span></span><span style="display:flex;"><span>        
</span></span><span style="display:flex;"><span>        <span style="color:#75715e"># Create a plot</span>
</span></span><span style="display:flex;"><span>        png_file <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">file.path</span>(<span style="color:#e6db74">&#34;./plots/&#34;</span>, <span style="color:#a6e22e">paste0</span>(gene_symbol, <span style="color:#e6db74">&#34;_plot.png&#34;</span>))  <span style="color:#75715e"># Generate the full file path</span>
</span></span><span style="display:flex;"><span>        <span style="color:#a6e22e">png</span>(png_file, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">800</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">600</span>)  <span style="color:#75715e"># Set width and height as needed</span>
</span></span><span style="display:flex;"><span>        <span style="color:#a6e22e">plotCounts</span>(dds <span style="color:#f92672">=</span> dds, gene <span style="color:#f92672">=</span> gene_id, main <span style="color:#f92672">=</span> gene_symbol)
</span></span><span style="display:flex;"><span>        <span style="color:#a6e22e">dev.off</span>()  <span style="color:#75715e"># Close the PNG device</span>
</span></span><span style="display:flex;"><span>        
</span></span><span style="display:flex;"><span>        <span style="color:#75715e"># Print a message to show progress</span>
</span></span><span style="display:flex;"><span>        <span style="color:#a6e22e">cat</span>(<span style="color:#e6db74">&#34;Plot saved for gene:&#34;</span>, gene_symbol, <span style="color:#e6db74">&#34;\n&#34;</span>)
</span></span><span style="display:flex;"><span>}
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Dispersion</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/dispersion_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">plotDispEsts</span>(dds, main <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;Dispersion Estimates&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># PCA</span>
</span></span><span style="display:flex;"><span>vsd <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">vst</span>(dds, blind<span style="color:#f92672">=</span><span style="color:#66d9ef">FALSE</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/PCA_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">plotPCA</span>(vsd, intgroup <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;condition&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># create histogram plot of p-values</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/Histo_P-values_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">hist</span>(res<span style="color:#f92672">$</span>padj, breaks<span style="color:#f92672">=</span><span style="color:#a6e22e">seq</span>(<span style="color:#ae81ff">0</span>, <span style="color:#ae81ff">1</span>, length <span style="color:#f92672">=</span> <span style="color:#ae81ff">21</span>), col <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;gray&#34;</span>, border <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;white&#34;</span>, 
</span></span><span style="display:flex;"><span>     xlab <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;&#34;</span>, ylab <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;&#34;</span>, main <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;Frequencies of padj-values&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># volcano plot</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/Volcano_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span>old.pal <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">palette</span>(<span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;#00BFFF&#34;</span>, <span style="color:#e6db74">&#34;#FF3030&#34;</span>)) <span style="color:#75715e"># low-hi colors</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">par</span>(mar<span style="color:#f92672">=</span><span style="color:#a6e22e">c</span>(<span style="color:#ae81ff">4</span>,<span style="color:#ae81ff">4</span>,<span style="color:#ae81ff">2</span>,<span style="color:#ae81ff">1</span>), cex.main<span style="color:#f92672">=</span><span style="color:#ae81ff">1.5</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">plot</span>(res<span style="color:#f92672">$</span>log2FoldChange, <span style="color:#f92672">-</span><span style="color:#a6e22e">log10</span>(res<span style="color:#f92672">$</span>padj), main<span style="color:#f92672">=</span><span style="color:#a6e22e">paste</span>(Groups[1], <span style="color:#e6db74">&#34;vs&#34;</span>, Groups[2]),
</span></span><span style="display:flex;"><span>     xlab<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;log2FC&#34;</span>, ylab<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;-log10(Padj)&#34;</span>, pch<span style="color:#f92672">=</span><span style="color:#ae81ff">20</span>, cex<span style="color:#f92672">=</span><span style="color:#ae81ff">0.5</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">with</span>(<span style="color:#a6e22e">subset</span>(res, padj<span style="color:#f92672">&lt;</span><span style="color:#ae81ff">0.05</span> <span style="color:#f92672">&amp;</span> <span style="color:#a6e22e">abs</span>(log2FoldChange) <span style="color:#f92672">&gt;=</span> <span style="color:#ae81ff">0</span>),
</span></span><span style="display:flex;"><span>     <span style="color:#a6e22e">points</span>(log2FoldChange, <span style="color:#f92672">-</span><span style="color:#a6e22e">log10</span>(padj), pch<span style="color:#f92672">=</span><span style="color:#ae81ff">20</span>, col<span style="color:#f92672">=</span>(<span style="color:#a6e22e">sign</span>(log2FoldChange) <span style="color:#f92672">+</span> <span style="color:#ae81ff">3</span>)<span style="color:#f92672">/</span><span style="color:#ae81ff">2</span>, cex<span style="color:#f92672">=</span><span style="color:#ae81ff">1</span>))
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">legend</span>(<span style="color:#e6db74">&#34;bottomleft&#34;</span>, title<span style="color:#f92672">=</span><span style="color:#a6e22e">paste</span>(<span style="color:#e6db74">&#34;Padj&lt;&#34;</span>, <span style="color:#ae81ff">0.05</span>, sep<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>), legend<span style="color:#f92672">=</span><span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;down&#34;</span>, <span style="color:#e6db74">&#34;up&#34;</span>), pch<span style="color:#f92672">=</span><span style="color:#ae81ff">20</span>,col<span style="color:#f92672">=</span><span style="color:#ae81ff">1</span><span style="color:#f92672">:</span><span style="color:#ae81ff">2</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># MD plot</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/MD_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">par</span>(mar<span style="color:#f92672">=</span><span style="color:#a6e22e">c</span>(<span style="color:#ae81ff">4</span>,<span style="color:#ae81ff">4</span>,<span style="color:#ae81ff">2</span>,<span style="color:#ae81ff">1</span>), cex.main<span style="color:#f92672">=</span><span style="color:#ae81ff">1.5</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">plot</span>(<span style="color:#a6e22e">log10</span>(res<span style="color:#f92672">$</span>baseMean), res<span style="color:#f92672">$</span>log2FoldChange, main<span style="color:#f92672">=</span><span style="color:#a6e22e">paste</span>(Groups[1], <span style="color:#e6db74">&#34;vs&#34;</span>, Groups[2]),
</span></span><span style="display:flex;"><span>     xlab<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;log10(mean of normalized counts)&#34;</span>, ylab<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;log2FoldChange&#34;</span>, pch<span style="color:#f92672">=</span><span style="color:#ae81ff">20</span>, cex<span style="color:#f92672">=</span><span style="color:#ae81ff">0.5</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">with</span>(<span style="color:#a6e22e">subset</span>(res, padj<span style="color:#f92672">&lt;</span><span style="color:#ae81ff">0.05</span> <span style="color:#f92672">&amp;</span> <span style="color:#a6e22e">abs</span>(log2FoldChange) <span style="color:#f92672">&gt;=</span> <span style="color:#ae81ff">0</span>),
</span></span><span style="display:flex;"><span>     <span style="color:#a6e22e">points</span>(<span style="color:#a6e22e">log10</span>(baseMean), log2FoldChange, pch<span style="color:#f92672">=</span><span style="color:#ae81ff">20</span>, col<span style="color:#f92672">=</span>(<span style="color:#a6e22e">sign</span>(log2FoldChange) <span style="color:#f92672">+</span> <span style="color:#ae81ff">3</span>)<span style="color:#f92672">/</span><span style="color:#ae81ff">2</span>, cex<span style="color:#f92672">=</span><span style="color:#ae81ff">1</span>))
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">legend</span>(<span style="color:#e6db74">&#34;bottomleft&#34;</span>, title<span style="color:#f92672">=</span><span style="color:#a6e22e">paste</span>(<span style="color:#e6db74">&#34;Padj&lt;&#34;</span>, <span style="color:#ae81ff">0.05</span>, sep<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>), legend<span style="color:#f92672">=</span><span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;down&#34;</span>, <span style="color:#e6db74">&#34;up&#34;</span>), pch<span style="color:#f92672">=</span><span style="color:#ae81ff">20</span>,col<span style="color:#f92672">=</span><span style="color:#ae81ff">1</span><span style="color:#f92672">:</span><span style="color:#ae81ff">2</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">abline</span>(h<span style="color:#f92672">=</span><span style="color:#ae81ff">0</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">palette</span>(old.pal) <span style="color:#75715e"># restore palette</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">## 3. Heatmap: sample-to-sample distance</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">### Generate distance matrix</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(pheatmap)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(RColorBrewer)
</span></span><span style="display:flex;"><span>sampleDists <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">dist</span>(<span style="color:#a6e22e">t</span>(<span style="color:#a6e22e">assay</span>(vsd)))
</span></span><span style="display:flex;"><span>sampleDistsMatrix <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as.matrix</span>(sampleDists)
</span></span><span style="display:flex;"><span><span style="color:#75715e">### Colorscheme</span>
</span></span><span style="display:flex;"><span>colors <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">colorRampPalette</span>(<span style="color:#a6e22e">rev</span>(<span style="color:#a6e22e">brewer.pal</span>(<span style="color:#ae81ff">9</span>,<span style="color:#e6db74">&#34;Blues&#34;</span>)))(<span style="color:#ae81ff">255</span>)
</span></span><span style="display:flex;"><span><span style="color:#75715e">### Generate heatmap</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/Heatmap_Smp2Smp_Distance_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">pheatmap</span>(sampleDistsMatrix, clustering_distance_rows<span style="color:#f92672">=</span>sampleDists,
</span></span><span style="display:flex;"><span>         clustering_distance_cols<span style="color:#f92672">=</span>sampleDists, col<span style="color:#f92672">=</span>colors)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">## 4. Heatmap of log transformed normal counts: top 10 genes.</span>
</span></span><span style="display:flex;"><span>top_genes <span style="color:#f92672">&lt;-</span> res<span style="color:#a6e22e">[order</span>(res<span style="color:#f92672">$</span>padj), ][1<span style="color:#f92672">:</span><span style="color:#ae81ff">10</span>,]
</span></span><span style="display:flex;"><span><span style="color:#75715e">### Get the gene names for the top 10</span>
</span></span><span style="display:flex;"><span>top_genes <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">row.names</span>(top_genes)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">### R log transformation</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/Heatmap_R_Log_Transformation_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span>rld <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">rlog</span>(dds, blind<span style="color:#f92672">=</span><span style="color:#66d9ef">FALSE</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">pheatmap</span>(<span style="color:#a6e22e">assay</span>(rld)[top_genes,], cluster_rows<span style="color:#f92672">=</span><span style="color:#66d9ef">FALSE</span>, show_rowname<span style="color:#f92672">=</span><span style="color:#66d9ef">TRUE</span>,cluster_cols<span style="color:#f92672">=</span><span style="color:#66d9ef">FALSE</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">## 5. Heatmap of Z scores</span>
</span></span><span style="display:flex;"><span>normalized_counts <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">counts</span>(dds,normalized<span style="color:#f92672">=</span><span style="color:#66d9ef">TRUE</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">write.csv</span>(normalized_counts, <span style="color:#e6db74">&#39;normalized_counts.csv&#39;</span>)
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>cal_z_score <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(x) {(x<span style="color:#f92672">-</span><span style="color:#a6e22e">mean</span>(x)) <span style="color:#f92672">/</span> <span style="color:#a6e22e">sd</span>(x)}
</span></span><span style="display:flex;"><span>zscore_all <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">t</span>(<span style="color:#a6e22e">apply</span>(normalized_counts, <span style="color:#ae81ff">1</span>, cal_z_score))
</span></span><span style="display:flex;"><span>zscore_subset <span style="color:#f92672">&lt;-</span> zscore_all[top_genes,]
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/Heatmap_Zscores_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">pheatmap</span>(zscore_subset)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">## 6. MA Plot</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/MA_plot.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">plotMA</span>(dds)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e">### MA plot with removed noises</span>
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">library</span>(apeglm)
</span></span><span style="display:flex;"><span>resLFC <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">lfcShrink</span>(dds, coef<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;condition_Disease_vs_Control&#34;</span>, type<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;apeglm&#34;</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">png</span>(<span style="color:#e6db74">&#34;./plots/MA_plot_removed_noises.png&#34;</span>, width <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, height <span style="color:#f92672">=</span> <span style="color:#ae81ff">1500</span>, res <span style="color:#f92672">=</span> <span style="color:#ae81ff">300</span>)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">plotMA</span>(resLFC)
</span></span><span style="display:flex;"><span><span style="color:#a6e22e">dev.off</span>()
</span></span></code></pre></div>
          </article>
        </div>
      </div>
      <div class="col-sm-12 col-md-12 col-lg-3">
        <div class="sticky-sidebar">
          
          <aside class="toc">
              <h5>
                Table Of Contents
              </h5>
              <div class="toc-content">
                <nav id="TableOfContents">
  <ul>
    <li><a href="#ensembl-ids-to-gene-symbols">ENSEMBL IDs to Gene Symbols</a></li>
    <li><a href="#plots">Plots</a></li>
  </ul>
</nav>
              </div>
          </aside>
          

          
          <aside class="tags">
            <h5>Tags</h5>
            <ul class="tags-ul list-unstyled list-inline">
              
              <li class="list-inline-item"><a href="https://rahriver.github.io/tags/r-lang" target="_blank">R Lang</a></li>
              
              <li class="list-inline-item"><a href="https://rahriver.github.io/tags/rna-seq-analysis" target="_blank">RNA-seq Analysis</a></li>
              
              <li class="list-inline-item"><a href="https://rahriver.github.io/tags/deseq2" target="_blank">DESeq2</a></li>
              
            </ul>
          </aside>
          

          
          <aside class="social">
            <h5>Social</h5>
            <div class="social-content">
              <ul class="list-inline">
                <li class="list-inline-item text-center">
                  <a target="_blank" href="https://twitter.com/share?text=My%20DESeq2%20Workflow&url=https%3a%2f%2frahriver.github.io%2fposts%2fdeseq2%2f">
                    <i class="fab fa-twitter"></i>
                  </a>
                </li>
                <li class="list-inline-item text-center">
                  <a target="_blank" href="https://api.whatsapp.com/send?text=My%20DESeq2%20Workflow: https%3a%2f%2frahriver.github.io%2fposts%2fdeseq2%2f">
                    <i class="fab fa-whatsapp"></i>
                  </a>
                </li>
                <li class="list-inline-item text-center">
                  <a target="_blank" href='mailto:?subject=My%20DESeq2%20Workflow&amp;body=Check%20out%20this%20site https%3a%2f%2frahriver.github.io%2fposts%2fdeseq2%2f'>
                    <i class="fa fa-envelope"></i>
                  </a>
                </li>
              </ul>
            </div>
          </aside>
          
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm-12 col-md-12 col-lg-9 p-4">
        
      </div>
    </div>
  </div>
  <button class="p-2 px-3" onclick="topFunction()" id="topScroll">
    <i class="fas fa-angle-up"></i>
  </button>
</section>


<div class="progress">
  <div id="scroll-progress-bar" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100"></div>
</div>
<Script src="/js/scrollProgressBar.js"></script>


<script>
  var topScroll = document.getElementById("topScroll");
  window.onscroll = function() {scrollFunction()};

  function scrollFunction() {
    if (document.body.scrollTop > 20 || document.documentElement.scrollTop > 20) {
      topScroll.style.display = "block";
    } else {
      topScroll.style.display = "none";
    }
  }

  function topFunction() {
    document.body.scrollTop = 0;
    document.documentElement.scrollTop = 0;
  }
</script>


<script src="/js/readingTime.js"></script>



  </div><footer>
    
 
 
<div class="container py-3" id="recent-posts">
    
    
    <div class="h3 text-center text-secondary py-3">
        Recent Posts
    </div>
    <div class="row justify-content-center">
        
        <div class="col-lg-4 col-md-6 pt-2">
            <div class="card h-100">
                
                <div class="card-header">
                    <a href="/posts/abs_sle/">
                        <img src="/images/posts/SLE-fig.png" class="card-img-top" alt="Abstract - Potential of Circulatory Cell-Free DNA">
                    </a>
                </div>
                
                <div class="card-body bg-transparent p-3 shadow-sm">
                    <a href="/posts/abs_sle/" class="primary-font card-title">
                        <h5 class="card-title bg-transparent" title="Abstract - Potential of Circulatory Cell-Free DNA">Abstract - Potential of ‚Ä¶</h5>
                    </a>
                    <div class="card-text secondary-font">
                        <p>A Systematic Review on the Potential of Circulatory Cell-Free DNA in the Diagnosis of Systemic Lupus Erythematosus Patients Ramin Yousefpour Shahrivar1, Zahra Ahmadi Fakhr2, Zahra Jahanbakhsh3, Fatemeh Karami4 Abstract Background: Systemic lupus erythematosus (SLE) is one of the frequent autoimmune ‚Ä¶</p>
                    </div>
                </div>
                <div class="mt-auto card-footer">
                    <span class="float-start">Sep 10, 2023</span>
                    <a href="/posts/abs_sle/" class="float-end btn btn-outline-info btn-sm">Read</a>
                </div>
            </div>
        </div>
        
        <div class="col-lg-4 col-md-6 pt-2">
            <div class="card h-100">
                
                <div class="card-header">
                    <a href="/posts/abs_aims/">
                        <img src="/images/posts/AIMS-fig.jpg" class="card-img-top" alt="Abstract - Detection of Chromosomal Abnormalities">
                    </a>
                </div>
                
                <div class="card-body bg-transparent p-3 shadow-sm">
                    <a href="/posts/abs_aims/" class="primary-font card-title">
                        <h5 class="card-title bg-transparent" title="Abstract - Detection of Chromosomal Abnormalities">Abstract - Detection of ‚Ä¶</h5>
                    </a>
                    <div class="card-text secondary-font">
                        <p>Detection of Major Structural Chromosomal Abnormalities in Human Karyotypes Using Convolutional Neural Networks Ramin Yousefpour Shahrivar1, Fatemeh Karami2 Abstract Background and aims: Chromosomal abnormalities are the prevalent cause of various types of genetic disorders, which can result in ‚Ä¶</p>
                    </div>
                </div>
                <div class="mt-auto card-footer">
                    <span class="float-start">Sep 10, 2023</span>
                    <a href="/posts/abs_aims/" class="float-end btn btn-outline-info btn-sm">Read</a>
                </div>
            </div>
        </div>
        
    </div>
</div>

<div class="text-center pt-2">
    
    <span class="px-1">
        <a href="https://github.com/rahriver/" aria-label="github">
            <svg xmlns="http://www.w3.org/2000/svg" width="2.7em" height="2.7em" viewBox="0 0 1792 1792">
                <path
                    d="M522 1352q-8 9-20-3-13-11-4-19 8-9 20 3 12 11 4 19zm-42-61q9 12 0 19-8 6-17-7t0-18q9-7 17 6zm-61-60q-5 7-13 2-10-5-7-12 3-5 13-2 10 5 7 12zm31 34q-6 7-16-3-9-11-2-16 6-6 16 3 9 11 2 16zm129 112q-4 12-19 6-17-4-13-15t19-7q16 5 13 16zm63 5q0 11-16 11-17 2-17-11 0-11 16-11 17-2 17 11zm58-10q2 10-14 14t-18-8 14-15q16-2 18 9zm964-956v960q0 119-84.5 203.5t-203.5 84.5h-224q-16 0-24.5-1t-19.5-5-16-14.5-5-27.5v-239q0-97-52-142 57-6 102.5-18t94-39 81-66.5 53-105 20.5-150.5q0-121-79-206 37-91-8-204-28-9-81 11t-92 44l-38 24q-93-26-192-26t-192 26q-16-11-42.5-27t-83.5-38.5-86-13.5q-44 113-7 204-79 85-79 206 0 85 20.5 150t52.5 105 80.5 67 94 39 102.5 18q-40 36-49 103-21 10-45 15t-57 5-65.5-21.5-55.5-62.5q-19-32-48.5-52t-49.5-24l-20-3q-21 0-29 4.5t-5 11.5 9 14 13 12l7 5q22 10 43.5 38t31.5 51l10 23q13 38 44 61.5t67 30 69.5 7 55.5-3.5l23-4q0 38 .5 103t.5 68q0 22-11 33.5t-22 13-33 1.5h-224q-119 0-203.5-84.5t-84.5-203.5v-960q0-119 84.5-203.5t203.5-84.5h960q119 0 203.5 84.5t84.5 203.5z" />

                <metadata>
                    <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
                        xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#" xmlns:dc="http://purl.org/dc/elements/1.1/">
                        <rdf:Description about="https://iconscout.com/legal#licenses"
                            dc:title="Github, Online, Project, Hosting, Square"
                            dc:description="Github, Online, Project, Hosting, Square" dc:publisher="Iconscout"
                            dc:date="2016-12-14" dc:format="image/svg+xml" dc:language="en">
                            <dc:creator>
                                <rdf:Bag>
                                    <rdf:li>Font Awesome</rdf:li>
                                </rdf:Bag>
                            </dc:creator>
                        </rdf:Description>
                    </rdf:RDF>
                </metadata>
            </svg>
        </a>
    </span>
    

    
    <span class="px-1">
        <a href="https://linkedin.com/in/ramin-yousefpour" aria-label="linkedin">
            <svg xmlns="http://www.w3.org/2000/svg" width='2.4em' height='2.4em' fill="#fff" aria-label="LinkedIn"
                viewBox="0 0 512 512">
                <rect width="512" height="512" fill="#0077b5" rx="15%" />
                <circle cx="142" cy="138" r="37" />
                <path stroke="#fff" stroke-width="66" d="M244 194v198M142 194v198" />
                <path d="M276 282c0-20 13-40 36-40 24 0 33 18 33 45v105h66V279c0-61-32-89-76-89-34 0-51 19-59 32" />
            </svg>
        </a>
    </span>
    

    

    

    
</div><div class="container py-4">
    <div class="row justify-content-center">
        <div class="col-md-4 text-center">
            <div class="pb-2">
                <a href="https://rahriver.github.io/" title="rahriver">
                    <img alt="Footer logo" src="/dna.png"
                        height="40px" width="40px">
                </a>
            </div>
            &copy; 2023  All rights reserved
            <div class="text-secondary">
                Made with
                <span class="text-danger">
                    &#10084;
                </span>
                and
                <a href="https://github.com/gurusabarish/hugo-profile" target="_blank"
                    title="Designed and developed by gurusabarish">
                    Hugo Profile
                </a>
            </div>
        </div>
    </div>
</div></footer><script src="/bootstrap-5/js/bootstrap.bundle.min.js"></script>
<script>
    if (document.body.className.includes("light")) {
        document.body.classList.add('dark');
        localStorage.setItem("pref-theme", 'dark');
    }
</script>
<script>
    document.getElementById("theme-toggle").addEventListener("click", () => {
        if (document.body.className.includes("dark")) {
            document.body.classList.remove('dark');
            localStorage.setItem("pref-theme", 'light');
        } else {
            document.body.classList.add('dark');
            localStorage.setItem("pref-theme", 'dark');
        }
    })

    var tooltipTriggerList = [].slice.call(document.querySelectorAll('[data-bs-toggle="tooltip"]'))
    var tooltipList = tooltipTriggerList.map(function (tooltipTriggerEl) {
        return new bootstrap.Tooltip(tooltipTriggerEl)
    })

</script>












  <section id="search-content" class="py-2">
    <div class="container" id="search-results"></div>
  </section>
</body>

</html>